frontend/views/containers/chatroom/NewDirectMessageModal.vue

```diff
@@ -137,37 +136,50 @@ export default ({
-          const lastMessageDate = this.ourUnreadMessages[chatRoomID].readUntil?.createdDate
-          return { chatRoomID, title, partners, lastJoinedPartner, picture, lastMessageDate }
+        .filter(chatRoomID => this.ourGroupDirectMessages[chatRoomID].visible)
+        .map(chatRoomID => {
+          const { title, partners, lastJoinedPartner, picture, lastMsgTimeStamp } = this.ourGroupDirectMessages[chatRoomID]
+          return { chatRoomID, title, partners, lastJoinedPartner, picture, lastMsgTimeStamp }
         })
         .sort((former, latter) => {
-          if (former.lastMessageDate > latter.lastMessageDate) {
-            return -1
-          } else if (former.lastMessageDate < latter.lastMessageDate) {
-            return 1
-          }
-          return former.title > latter.title ? 1 : -1
+          const diff = former.lastMsgTimeStamp - latter.lastMsgTimeStamp
+          return diff > 0 ? -1 : (diff < 0 ? 1 : (former.title > latter.title ? 1 : -1))
         })
     },
     filteredRecents () {
+      if (!this.searchText && !this.selections.length) {
+        return this.ourRecentConversations
+      }
       return this.ourRecentConversations.filter(({ title, partners }) => {
+        const partnerIDs = partners.map(p => p.contractID)
         const upperCasedSearchText = String(this.searchText).toUpperCase().normalize()
-        if (!difference(partners, this.selections).length) {
+        if (!difference(partnerIDs, this.selections).length) {
+          // match with contractIDs
           return false
-        } else if (String(title).toUpperCase().normalize().indexOf(upperCasedSearchText) > -1) {
-          return true
-        } else if (String(partners.join(', ')).toUpperCase().indexOf(upperCasedSearchText) > -1) {
+        } else if (String(title).toUpperCase().normalize().includes(upperCasedSearchText)) {
+          // match with title
           return true
+        } else {
+          // match with username and displayname
+          const userKeywords = upperCasedSearchText.replace(/\s/g, '').split(',')
+          return userKeywords.reduce((found, userKeyword, index, arr) => {
+            const isLastUserKeyword = index === arr.length - 1
+            let currentFound = false
+            if (isLastUserKeyword) {
+              currentFound = partners.findIndex(p => {
+                return p.username.toUpperCase().normalize().includes(userKeyword) ||
+                  p.displayName.toUpperCase().normalize().includes(userKeyword)
+              }) >= 0
+            } else {
+              currentFound = partners.findIndex(p => {
+                return p.username.toUpperCase().normalize() === userKeyword ||
+                  p.displayName.toUpperCase().normalize() === userKeyword
+              }) >= 0
+            }
+            return found && currentFound
+          }, true)
         }
-        return false
-      })
+      }).sort((a, b) => a.partners.length > b.partners.length ? 1 : -1)
```

Comment:
What are these changes for, and which issue are they related to?

Reply:
> What are these changes for, and which issue are they related to?

This change is for the third stuff of this PR summary.
`Improved to search DMs by keyword (Possible to use usernames, and displayNames)`

Reply:
While I was testing the searching DMs, I noticed that in case I tried to search DMs by keyword, I can't get the DMs that I wanted to search.

Example cases;
Let's imagine that we have four users; alexjin(Silver-IT), greg(Taoeffect), sebin(Sebin Song), andrea
And also we have three DMs; `Taoeffect`, `Sebin Song, Taoeffect, andrea`, `Sebin Song`

- We couldn't find any DMs using the keyword `greg` since the keyword `greg` doesn't exist in the title of any DMs
- We couldn't find any DMs using the keyword `andrea, Taoeffect` since the keyword `andrea, Taoeffect` doesn't exist in the title of any DMs
- When we type keyword `Sebin Song`, it displays two DMs; `Sebin Song, Taoeffect, andrea` and `Sebin Song` in order. And when I press Enter, the first DM was opened. But I thought the second DM(`Sebin Song`) would be opened because I only typed keyword `Sebin Song`